camillewalters/Eggstravaganza
# Goals Interaction

## Added
- Added `Colliders` to Goals and enabled `isTrigger`
- Added values for items (golden eggs=3, eggs=1, food=0, etc.) to keep score, can be edited in the editor
- Matches goal ID to player ID
- Items get destroyed and `eggInventory` gets emptied when goal is reached
- **Added EggBehavior to Tomato Prefab to allow player's to pick it up and throw, mostly for testing purposes, not sure if we want another ItemBehavior?**

## To Test
1. Open the `StephArena` scene
2. Uncomment `EggSpawner` to have items spawn
3. Start the host and add players (start clients)
4. From left-right-top-bottom, goal ids go from 0-3. Eg. First player to spawn should have id=0 and so their goal is top left. Second player has id=1, their goal is top right.

## Notes
I decided that when the player reaches the goal, all items get destroyed even non-eggs because their positions would need to be updated. (eg. All eggs disappear, and non-egg items are still above the head with gaps in between where the eggs were). Maybe we can do differently in another PR?

Closes #28
